File: core-exchange/src/main/java/org/exchange/internal/app/core/utils/CurrencyUtils.java
package org.exchange.internal.app.core.utils;

import static org.exchange.app.common.api.model.Direction.BUY;

import org.exchange.app.common.api.model.Direction;
import org.exchange.app.common.api.model.Pair;

public class CurrencyUtils {

  public static String pairToCurrency(final Pair currencyChange, final Direction direction) {
    if (BUY.equals(direction)) {
      return currencyChange.name().split("_")[1];
    } else {
      return currencyChange.name().split("_")[0];
    }
  }
}

File: frontend-client/src/app/utils/currency-utils.ts
import { UserTicket } from '../api/model/userTicket';
import { PairUtils } from './pair-utils';

export class CurrencyUtils {
  public static ticketToCurrency(ticket: UserTicket | null): string {
    if (ticket == null) {
      return '';
    }
    if (ticket.direction == 'BUY') {
      return PairUtils.getQuoteCurrency(ticket.pair);
    } else {
      return PairUtils.getBaseCurrency(ticket.pair);
    }
  }
}
